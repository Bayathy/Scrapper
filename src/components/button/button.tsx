import styled from "@emotion/styled";
import type { FC } from "react";
import { Component } from "../../types/components";
import { css, keyframes, useTheme } from "@emotion/react";

export type IconButtonProps = {
  imageSrc?: string;
} & Component<"button">;

export const StyledButton = styled.button`
  border: none;
  background: none;
  cursor: pointer;
  font-size: 1.5rem;
  padding: 0.5rem 2rem;
  background: white;
  border-radius: 20px;
  :hover {
    background-color: gray;
  }
`;


export const IconButton: FC<IconButtonProps> = ({
  cx,
  imageSrc,
  onClick,
  ...rest
}) => {
  const theme = useTheme()

  const bounce = keyframes`
  from {
    outline: 0px solid ${theme.color.secondary};
    outline-offset: -4px;
  }
  to {
    outline: 4px solid ${theme.color.secondary};
    outline-offset: -4px;
  }
`;

  return (
    <button
      onClick={onClick}
      {...rest}
      css={[
        css`
          box-sizing: border-box;
          border: none;
          background: none;
          width: 3rem;
          height: 3rem;
          border-radius: 999px;
          padding: 0;
          :hover {
            animation: ${bounce} 0.2s;
            outline: 4px solid ${theme.color.secondary};
            outline-offset: -4px;
          }
        `,
        cx,
      ]}
    >
      <img
        css={css`
          width: inherit;
          aspect-ratio: 1/1;
          border-radius: 999px;
        `}
        src={imageSrc ?? ""}
        alt="your Icon"
      />
    </button>
  );
};
