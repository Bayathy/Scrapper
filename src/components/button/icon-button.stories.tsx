import { IconButton } from "./button";

import type { IconButtonProps } from "./button";
import type { Meta, StoryObj } from "@storybook/react";


const meta: Meta<IconButtonProps> = {
  title: "Button/IconButton",
  component: IconButton,
};

export default meta;

export const Default: StoryObj<IconButtonProps> = {
  render: () => (
    <IconButton imageSrc="https://images.ctfassets.net/hrltx12pl8hq/01rJn4TormMsGQs1ZRIpzX/02e9885a9ae69312da844bc58eedced1/Artboard_Copy_22.png?fit=fill&w=600&h=400" />
  ),
};
