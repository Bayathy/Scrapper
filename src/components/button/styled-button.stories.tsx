import React from "react";

import { StyledButton } from "./button";

import type { Meta, StoryObj } from "@storybook/react";


const meta: Meta<typeof StyledButton> = {
  title: "Button/StyledButton",
  component: StyledButton,
};

export default meta;

export const Default: StoryObj<typeof StyledButton> = {
  render: () => <StyledButton>Button</StyledButton>,
};
