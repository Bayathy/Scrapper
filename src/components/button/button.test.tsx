import { composeStories } from "@storybook/testing-react";
import * as ButtonStories from "./button.stories";
import { render, screen, within } from "@testing-library/react";
import React from "react";
import { StyledButton } from "./button";

const { Default } = composeStories(ButtonStories);

describe("testing button component", () => {
  it("take snapshot", () => {
    const { asFragment } = render(<Default>Button</Default>);
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
  });
  it("test children state", () => {
    render(<StyledButton>Button</StyledButton>);
    const buttonElement = screen.getByText("Button");
    expect(buttonElement).not.toBeNull();
  });
});
