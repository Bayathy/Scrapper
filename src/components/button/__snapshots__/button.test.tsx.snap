// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing button component take snapshot 1`] = `
<DocumentFragment>
  <button
    class="css-8tgcgv-StyledButton e1vo3cka0"
  >
    Button
  </button>
</DocumentFragment>
`;
